### ISSUE TITLE
Unable to save interfaces - SoundBlossomer was not able to save/apply your changes.

### ISSUE THREAD

**Issue by snrkl** [LucaTNT/SoundBlossomer#1]
Tried creating an additional interface as per your [blog post](https://lucatnt.com/2013/10/introducing-soundblossomer/) - I even tried using the names in your screenshot in case there was some silly naming convention that had to be followed.

When I click save and reload, I receive the following error:

> Unable to save interfaces
> SoundBlossomer was not able to save/apply your changes.

If I reboot, sound flower doesn't load (it looks like `/System/Library/Extensions/Soundflower.kext/Contents/Info.plist` is edited, I checked) but until I remove the newly created interfaces, soundflower won't load.

System specifics:
OSX: 10.10.4 (14E46)
SoundFlower: 1.6.6b

**Comment by snrkl** [comment]
Some more investigation - console log shows:

> 20/07/2015 4:14:04.833 pm com.apple.kextd[44]: ERROR: invalid signature for com.Cycling74.driver.Soundflower, will not load

**Comment by snrkl** [comment]
More info - based on a post [here](https://github.com/crmne/Soundflower/issues/4), I tried

> sudo nvram boot-args="kext-dev-mode=1"

manually edited the plist file so the  for the sound devices I wanted, rebooted and it seems I now have several 2channel devices...

Now running a system that allows unsigned kexts to load is not ideal.. any ideas on how this could be addressed?

**Comment by LucaTNT** [comment]
TL;DR: you'll have to keep kext dev mode enabled. 

You got to the core of the problem: Yosemite enforces kext signatures, and with El Capitan things will become even trickier with rootless, another protection that prevents app from writing to many system folders, and of course /System/Library/Extensions/ is one of them. 

Sadly I doubt there's any viable solution other than enabling kext dev mode as you just did. Editing the Info.plist changes the kext's signature, making the current one (if present) invalid, so it would have to be re-signed every time SoundBlossomer makes any change to that file, which would not be possible even if I had the special certificate required to sign kexts (a regular developer cert is not enough, you have to make a special request to Apple to get an appropriate one), since of course I can't ship the certificate itself with SoundBlossomer, and without the certificate the app wouldn't be able to sign the kext.
